crates/commitlog/src/lib.rs

```diff
@@ -390,3 +397,85 @@ impl<T: Send + Sync + 'static> Commitlog<T> {
         rx
     }
 }
+
+/// Obtain an iterator which traverses the commitlog located at the `root`
+/// directory from the start, yielding [`Commit`]s.
+///
+/// Starts the traversal without the upfront I/O imposed by [`Commitlog::open`].
+/// See [`Commitlog::commits`] for more information.
+pub fn commits(root: impl Into<PathBuf>) -> io::Result<impl Iterator<Item = Result<Commit, error::Traversal>>> {
+    commits_from(root, 0)
+}
+
+/// Obtain an iterator which traverses the commitlog located at the `root`
+/// directory starting from `offset` and yielding [`Commit`]s.
+///
+/// Starts the traversal without the upfront I/O imposed by [`Commitlog::open`].
+/// See [`Commitlog::commits_from`] for more information.
+pub fn commits_from(
+    root: impl Into<PathBuf>,
+    offset: u64,
+) -> io::Result<impl Iterator<Item = Result<Commit, error::Traversal>>> {
+    commitlog::commits_from(repo::Fs::new(root), DEFAULT_LOG_FORMAT_VERSION, offset)
+}
```

Comment:
Is the only difference between these and the commitlog:: variants the abstraction over the repo?

Would it be better to just have the ones in commitlog for this and let the caller call `repo::Fs::new` manually?

(In my case, a custom TarRepo is used to consume the logs without extracting them from the *.tar archive first)

Reply:
Most of what you’re using isn’t exported from the crate. We can change that, but I’ll probably be molested to add docstrings everywhere 😿